validator/client/validator.go

```diff
@@ -1128,6 +1150,12 @@ func (v *validator) buildSignedRegReqs(ctx context.Context, pubkeys [][fieldpara
 			continue
 		}
 
+		// map is populated before this function in buildPrepProposerReq
+		_, ok := v.pubkeyToValidatorIndex[k]
+		if !ok {
+			continue
+		}
```

Comment:
Do you need a read lock to access this map?

Reply:
not currently, is this something that should have a read lock and write lock for updating?

Reply:
access to this map should be sequential only so read and writes shouldn't overlap in this case I think we won't need the lock but will test.